src/main/scala/no/ndla/audioapi/service/WriteService.scala

```diff
@@ -0,0 +1,83 @@
+package no.ndla.audioapi.service
+
+import java.io.ByteArrayInputStream
+
+import com.typesafe.scalalogging.LazyLogging
+import no.ndla.audioapi.model.api._
+import no.ndla.audioapi.model.domain.Audio
+import no.ndla.audioapi.repository.AudioRepository
+import no.ndla.audioapi.service.search.SearchIndexService
+import org.scalatra.servlet.FileItem
+import scala.util.{Failure, Random, Success, Try}
+import java.lang.Math.max
+
+
+trait WriteService {
+  this: ConverterService with ValidationService with AudioRepository with SearchIndexService with AudioStorageService =>
+  val writeService: WriteService
+
+  class WriteService extends LazyLogging {
+    def storeNewAudio(newAudioMeta: NewAudioMetaInformation, files: Seq[FileItem]): Try[AudioMetaInformation] = {
+      val fileValidationMessages = files.flatMap(validationService.validateAudioFile)
+      if (fileValidationMessages.nonEmpty) {
+        return Failure(new ValidationException(errors=fileValidationMessages))
+      }
+
+      for {
+        uploadedAudios <- uploadFiles(files, newAudioMeta.audioFiles)
+        domainAudio <-  Try(converterService.toDomainAudioMetaInformation(newAudioMeta, uploadedAudios))
+        _ <- validationService.validate(domainAudio)
+        r <- Try(audioRepository.insert(domainAudio))
+        _ <- searchIndexService.indexDocument(r)
+      } yield converterService.toApiAudioMetaInformation(r)
+    }
+
+    private[service] def uploadFiles(filesToUpload: Seq[FileItem], audioFileMetas: Seq[NewAudioFile]): Try[Seq[Audio]] = {
```

Comment:
Splittet den opp i to funksjoner, som forhåpentligvis gjør det litt mer lesbart.